shared/fs/container.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import * as React from 'react'
+import * as I from 'immutable'
+import {namedConnect, type RouteProps} from '../util/container'
+import {navigateAppend, navigateUp} from '../actions/route-tree'
+import * as Constants from '../constants/fs'
+import * as Types from '../constants/types/fs'
+import {isMobile} from '../constants/platform'
+import Folder from './folder/container'
+import {NormalPreview} from './filepreview'
+import Loading from './common/loading'
+
+const mapStateToProps = state => ({
+  _pathItems: state.fs.pathItems,
+})
+
+const mapDispatchToProps = (dispatch, {routePath}) => ({
+  _emitBarePreview: (path: Types.Path) => {
+    dispatch(navigateUp()) // pop this route node before appending barePreview
+    dispatch(navigateAppend([{props: {path}, selected: 'barePreview'}]))
+  },
+})
+
+const mergeProps = (stateProps, dispatchProps, {routeProps, routePath}) => {
+  const path = routeProps.get('path', Constants.defaultPath)
+  const isDefinitelyFolder = Types.getPathElements(path).length <= 3
+  const pathItem = stateProps._pathItems.get(path, Constants.unknownPathItem)
+  return {
+    emitBarePreview: () => dispatchProps._emitBarePreview(path),
+    mimeType: !isDefinitelyFolder && pathItem.type === 'file' ? pathItem.mimeType : null,
+    path,
+    pathType: isDefinitelyFolder ? 'folder' : stateProps._pathItems.get(path, Constants.unknownPathItem).type,
+    routePath,
+  }
+}
+
+type ChooseComponentProps = {
+  emitBarePreview: () => void,
+  mimeType: ?Types.Mime,
+  path: Types.Path,
+  pathType: Types.PathType,
+  routePath: I.List<string>,
+}
+
+const useBare = isMobile
+  ? (mimeType: ?Types.Mime) => {
+      return Constants.viewTypeFromMimeType(mimeType) === 'image'
+    }
+  : (mimeType: ?Types.Mime) => {
+      return false
+    }
+
+class ChooseComponent extends React.PureComponent<ChooseComponentProps> {
+  componentDidMount() {
```

Comment:
nit: you can use the `componentDidMount = () => {` style here and the functions below